core/src/context.rs

```diff
@@ -246,4 +250,59 @@ impl Context {
 
         Ok(())
     }
+
+    pub fn spawn<F>(&self, target: F) where F: 'static + Send + Fn(&mut ContextProxy) {
+        let mut cxp = ContextProxy {
+            current: self.current,
+            event_proxy: self.event_proxy.as_ref().map(|p| p.make_clone())
+        };
+
+        std::thread::spawn(move || target(&mut cxp));
+    }
+}
+
+/// A bundle of data representing a snapshot of the context when a thread was spawned. It supports
+/// a small subset of context operations. You will get one of these passed to you when you create a
+/// new thread with `cx.spawn()`.
+pub struct ContextProxy {
+    pub current: Entity,
+    pub event_proxy: Option<Box<dyn EventProxy>>,
+}
+
+pub enum ProxyEmitError {
+    Unsupported,
+    EventLoopClosed,
+}
```

Comment:
This should probably implement `std::error::Error`.

Reply:
Done.